pkg/verifier/aws/aws_verifier.go

```diff
@@ -339,11 +340,30 @@ func (a *AwsVerifier) createEC2Instance(input createEC2InstanceInput) (string, e
 
 	// Wait up to 5 minutes for the instance to be running
 	waiter := ec2.NewInstanceRunningWaiter(a.AwsClient)
+
 	if err := waiter.Wait(input.ctx, &ec2.DescribeInstancesInput{InstanceIds: []string{instanceID}}, 2*time.Minute); err != nil {
+
+		waiterErr := fmt.Errorf("%s: terminated %s after timing out waiting for instance to be running", err, instanceID)
+
+		resp, err := a.AwsClient.DescribeInstances(context.TODO(), &ec2.DescribeInstancesInput{
+			InstanceIds: []string{instanceID},
+		})
+		if err != nil {
+			fmt.Println("Warning: Waiter Describe instances failure.")
+		}
+
+		stateCode := *resp.Reservations[0].Instances[0].StateReason.Code
+		stateMessage := *resp.Reservations[0].Instances[0].StateReason.Message
```

Comment:
Each of these has the potential to cause a nil pointer. We should check they are not nil before getting the value, with something like this

```go
 var stateCode, stateMessage string
 if resp.Reservations[0].Instances[0].StateReason.Code != nil {
   stateCode = *resp.Reservations[0].Instances[0].StateReason.Code
 }
 if resp.Reservations[0].Instances[0].StateReason.Message != nil {
   stateMessage = *resp.Reservations[0].Instances[0].StateReason.Message
 }
```

Reply:
Done